src/Microsoft.Azure.EventHubs/Primitives/ClientEntity.cs

```diff
@@ -55,6 +65,43 @@ public RetryPolicy RetryPolicy
         /// <returns>The asynchronous operation</returns>
         public abstract Task CloseAsync();
 
+        /// <summary>
+        /// Registers a <see cref="EventHubsPlugin"/> to be used with this client.
+        /// </summary>
+        public virtual void RegisterPlugin(EventHubsPlugin eventHubPlugin)
+        {
+            if (eventHubPlugin == null)
+            {
+                throw new ArgumentNullException(nameof(eventHubPlugin), Resources.ArgumentNullOrWhiteSpace.FormatForUser(nameof(eventHubPlugin)));
+            }
+            if (this.RegisteredPlugins.Any(p => p.Key == eventHubPlugin.Name))
+            {
+                throw new ArgumentException(nameof(eventHubPlugin), Resources.PluginAlreadyRegistered.FormatForUser(nameof(eventHubPlugin.Name)));
```

Comment:
>nameof [](start = 116, length = 6)

just plugin.Name, right?

Reply:
You mean this instead ? 

```C#
if (this.RegisteredPlugins.Any(p => p.Value.Name == eventHubPlugin.Name))
            {
                throw new ArgumentException(nameof(eventHubPlugin), Resources.PluginAlreadyRegistered.FormatForUser(nameof(eventHubPlugin.Name)));
            }
```

Reply:
nameof will return 'Name' and the error message will build as: The Name plugin has already been registered.

---
In reply to: [224288914](https://github.com/Azure/azure-event-hubs-dotnet/pull/324#discussion_r224288914) [](ancestors = 224288914)

Reply:
Fair enought